src/controller/user.controller.ts

```diff
@@ -1,21 +1,39 @@
 import { Server, Socket } from "socket.io";
+import jwt from 'jsonwebtoken';
 
 export class UserController {
     private io: Server;
+    private secretKey: string;
 
-    constructor(io : Server) {
+    constructor(io : Server, secretKey : string) {
         this.io = io;
+        this.secretKey = secretKey;
         this.initializeChat();
     }
 
     private initializeChat(): void {
         this.io.on('connection', (socket: Socket) => {
-            console.log('A user connected', socket.id);
+            console.log('A user connected by socketID:', socket.id);
+
+            const token = socket.handshake.headers['authorization'];
+            var userID: any;
+            if(token) {
+                try {
+                    const decodedToken: any = jwt.verify(token, this.secretKey);
+                    userID = decodedToken.id;
+                    console.log('User connected with userID:', userID);
+                } catch (error) {
+                    console.error('Error decoding token:', error);
+                }
+            } else {
+                console.error('No token provided');
+                socket.disconnect();
```

Comment:
ตรงนี้ก่อนจะ disconnect ให้
`socket.emit('auth error', 'Error token not provided');` ก่อน client จะได้รู้